src/components/AvatarWithImagePicker.js

```diff
@@ -148,17 +147,18 @@ class AvatarWithImagePicker extends React.Component {
     }
 
     /**
-     * Check if the attachment resolution is bigger than required.
+     * Check if the attachment resolution matches constraints.
      *
-     * @param {String} imageUri
+     * @param {Object} image
      * @returns {Promise}
      */
-    isValidResolution(imageUri) {
-        return new Promise((resolve) => {
-            Image.getSize(imageUri, (width, height) => {
-                resolve(height >= CONST.AVATAR_MIN_HEIGHT_PX && width >= CONST.AVATAR_MIN_WIDTH_PX);
-            });
-        });
+    isValidResolution(image) {
+        const isResolutionWithinRange = resolution => (
+            (resolution.height >= CONST.AVATAR_MIN_HEIGHT_PX && resolution.width >= CONST.AVATAR_MIN_WIDTH_PX)
+            && (resolution.height <= CONST.AVATAR_MAX_HEIGHT_PX && resolution.width <= CONST.AVATAR_MAX_WIDTH_PX)
+        );
+
+        return getImageResolution(image).then(isResolutionWithinRange);
```

Comment:
Can you please update this code so that there is no `isResolutionWithinRange` variable? The method can be defined directly in `then()` and it is easier to read and follow.

Reply:
Sure, fixed.